corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -314,52 +311,25 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
             usedWidth = 0;
         }
 
-        function pushChildAndResetGroup(child) {
-            newChildren.push(child);
-            resetCurrentGroup();
-        }
-
-        function groupElementsAndResetGroup(child) {
-            const newGroup = Container.groupElements(child, elementType);
-            newChildren.push(newGroup);
-            resetCurrentGroup();
-        }
-
-        function addToGroupForAllowedWidth(child) {
-            const questionTileWidth = Question.calculateColumnWidthForPerRowStyle(child.style);
-            usedWidth += questionTileWidth;
-            if (usedWidth > constants.GRID_COLUMNS) {
-                resetCurrentGroup();
-                usedWidth += questionTileWidth;
-            }
-            addToCurrentGroup(child);
-        }
-
-        function processChildForGroupingQuestions(child) {
-            if (child.type === constants.QUESTION_TYPE) {
-                addToGroupForAllowedWidth(child);
-            } else if (child.type === constants.GROUP_TYPE || child.type === constants.REPEAT_TYPE) {
-                groupElementsAndResetGroup(child);
-            } else {
-                pushChildAndResetGroup(child);
-            }
-        }
-
-        function processChildForGroupingGroups(child) {
-            if (child.type === constants.GROUP_TYPE) {
-                addToGroupForAllowedWidth(child);
+        for (let child of json.children) {
+            if (child.type === constants.QUESTION_TYPE || child.type === constants.GROUP_TYPE) {
+                const elementTileWidth = Question.calculateColumnWidthForPerRowStyle(child.style);
```

Comment:
`calculateColumnWidthForPerRowStyle` should be defined somewhere else since it's no longer specific to `Question`. Maybe it can be a part of `GroupedElementTileRow` and renamed to something like `calculateElementWidth`

Reply:
https://github.com/dimagi/commcare-hq/pull/33834/commits/50eec9c7bdb4113e5657009f9f4adc11d5c058c3